Mathlib/Topology/ShrinkingLemma.lean

```diff
@@ -240,3 +242,47 @@ theorem exists_iUnion_eq_closed_subset (uo : ∀ i, IsOpen (u i)) (uf : ∀ x, {
     ∃ v : ι → Set X, iUnion v = univ ∧ (∀ i, IsClosed (v i)) ∧ ∀ i, v i ⊆ u i :=
   let ⟨v, vU, hv⟩ := exists_subset_iUnion_closed_subset isClosed_univ uo (fun x _ => uf x) uU.ge
   ⟨v, univ_subset_iff.1 vU, hv⟩
+
+end NormalSpace
+
+section T2LocallyCompactSpace
+
+variable {u : ι → Set X} {s : Set X} [T2Space X] [LocallyCompactSpace X]
+
+lemma exists_subset_closure_subset_t2space (hs : IsCompact s) (U : Set X) (Uo : IsOpen U)
```

Comment:
Is it just [exists_open_between_and_isCompact_closure](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Separation.html#exists_open_between_and_isCompact_closure)?

Reply:
oh yes, I didn't notice that.

What do you think of the sketch of a proof of a shrinking lemma below? I would have to use an induction over a finite set, for which I would have to find the right APIs...

Reply:
done with the shrinking lemma.

Reply:
@fpvandoorn I tried to make `PartitionOfFunction` through `BumpCovering`. The current problem is that [BumpCovering.exists_isSubordinate_of_locallyFinite_of_prop](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/PartitionOfUnity.html#BumpCovering.exists_isSubordinate_of_locallyFinite_of_prop) depends on `NormalSpace X` through [exists_subset_iUnion_closure_subset](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/ShrinkingLemma.html#exists_subset_iUnion_closure_subset). While I could write a variation of [ShrinkingLemma.PartialRefinement.exists_gt](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/ShrinkingLemma.html#ShrinkingLemma.PartialRefinement.exists_gt), I don't know how to assure that the maximal element has relatively compact open covers, except duplicating all the API's with that condition. How should I proceed?

Reply:
@urkud @fpvandoorn I think I'm done with `PartitionOfUnity`.